### ISSUE TITLE
Logic gets stuck at preAlignSegments()

### ISSUE THREAD

**Issue by bgeorge0** [SlicerRt/SegmentRegistration#1]
I am having trouble using this module as it gets stuck during the 'preAlignSegments' function for an unknown reason. I have done some digging into the code, but with no luck.

I have tried to find the problem by running the module from the python interactor.

slicer.segmentRegistrationLogic.cropMovingVolumes() works fine however, slicer.segmentRegistrationLogic.prealignSegments just hangs 3D Slicer. I added a logging.info call at the start of this function but it never gets executed, so I'm lost as to how to investigate this further.

There also seem to be some bugs in the 'Run self test' code. Firstly, the scope of self.delayMs seems to be incorrect, as it is not declared when the test is run. I worked around this by replacing all the instances with 700. It now gets stuck with:
slicer.util.MRMLNodeNotFoundException: could not find nodes in the scene by name or id 'Moving_Structure_Padded-surface'

This is in Slicer 4.10.1 on MacOS

**Comment by lassoan** [comment]
Does it work well if you install this (and all required additional extensions) using Extension Manager and use the module using the Segment Registration module GUI?

You can try first on some simple data sets (load MRHead sample data and draw  two similar 3D shapes in two segmentations) and see if you have any problems with those. If you do, then send those data sets so that we can have a look.

**Comment by bgeorge0** [comment]
I installed it via Segmentation manager already, so that is not the issue.

I have narrowed the problem down a little. The problem occurs when performing harden transform on the pre-aligned segmentation. I get the same problem if I just apply a transformation to the segmentation and then try and harden it, regardless of whether I use this extension. Therefore, I don't think that this issue is to do with this extension, but something in 3D Slicer.

I've just posted about this in the 3D Slicer forum as well - think there might be some crossing of the streams soon.
